repo/packages/S/spark/0/marathon.json

```diff
@@ -0,0 +1,38 @@
+{
+    "id": "{{spark.framework-name}}",
+    "cpus": {{spark.cpus}},
+    "mem": {{spark.mem}},
+    "instances": {{spark.instances}},
+    "cmd": "echo spark.mesos.principal={{mantl.mesos.principal}} >> ./conf/spark-defaults.conf && echo spark.mesos.secret={{mantl.mesos.secret}} >> ./conf/spark-defaults.conf && mv ./conf/log4j-dispatcher.properties ./conf/log4j.properties && ./bin/spark-class org.apache.spark.deploy.mesos.MesosClusterDispatcher --port $PORT0 --webui-port $PORT1 --master {{mesos.master}} --zk {{spark.zookeeper}} --host {{spark.host}} --name {{spark.framework-name}}",
```

Comment:
The framework runs great on clusters that have authentication enabled - which is the default. However, it will not work on clusters where mesos auth has been disabled. With this line, `./conf/spark-defaults.conf` is generated even on a cluster that doesn't need the credentials. Spark thinks it must authenticate, though, and is never able to fully register itself as a framework.

There is a template helper that should make this an easy fix. If you wrap the commands that write the credentials with `mantl.mesos.authentication-enabled`, it will only include them on clusters with mesos auth turned on. Something like this (untested):

```
"cmd": "{{#mantl.mesos.authentication-enabled}}echo spark.mesos.principal={{mantl.mesos.principal}} >> ./conf/spark-defaults.conf && echo spark.mesos.secret={{mantl.mesos.secret}} >> ./conf/spark-defaults.conf && {{/mantl.mesos.authentication-enabled}}mv ./conf/log4j-dispatcher.properties ./conf/log4j.properties && ./bin/spark-class org.apache.spark.deploy.mesos.MesosClusterDispatcher --port $PORT0 --webui-port $PORT1 --master {{mesos.master}} --zk {{spark.zookeeper}} --host {{spark.host}} --name {{spark.framework-name}}",
```